src/D2L.CodeStyle.Analyzers/Immutability/TypeDeclarationImmutabilityAnalyzer.cs

```diff
@@ -62,6 +63,14 @@ ImmutabilityContext immutabilityContext
 				return;
 			}
 
+			// We want to run our analyzer to run for generated code too, but
+			// the generated code from resx files contain static variables
+			// which don't contribute to mutability but can't be analyzed as
+			// safe (e.g. CultureInfos, only some of those are read-only).
+			if( IsFromGeneratedResourcesFile( ctx.Symbol ) ) {
+				return;
+			}

```

Comment:
https://github.com/Brightspace/D2L.CodeStyle/blob/master/src/D2L.CodeStyle.Analyzers/Immutability/UnsafeStaticsAnalyzer.cs#L26-L32

This is less crazy than what we're doing now. I'm open to other ideas in the future.